### PR TITLE
Fixes because of PHP7 strict mode

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/DI/Config/Adapters/NeonAdapter.php [hunk 1]
```diff
@@ -37,20 +37,21 @@ private function process(array $arr)
...
-				$val[Helpers::EXTENDS_KEY] = Helpers::OVERWRITE;
-
-			} elseif (preg_match('#^(\S+)\s+' . self::INHERITING_SEPARATOR . '\s+(\S+)\z#', $key, $matches)) {
-				if (!is_array($val) && $val !== NULL) {
-					throw new Nette\InvalidStateException("Inheritance operator is available only for arrays, item '$key' is not array.");
-				}
-				list(, $key, $val[Helpers::EXTENDS_KEY]) = $matches;
-				if (isset($res[$key])) {
-					throw new Nette\InvalidStateException("Duplicated key '$key'.");
+			if (is_string($key)) {
```
Comment (dg) [inline src/DI/Config/Adapters/NeonAdapter.php]: Please can you change the deep of nesting, ie. use `is_string($key)` in both conditions or `if (!is_string($key)))` as first condition?

**Comment by kukulich** [comment]
Forcepushed.

File: src/DI/ContainerBuilder.php [hunk 2]
```diff
@@ -266,7 +266,9 @@ public function autowireArguments($class, $method, array $arguments)
 		if (!$rm->isPublic()) {
 			throw new ServiceCreationException("$class::$method() is not callable.");
 		}
-		$this->addDependency($rm->getFileName());
+		if (!$rm->isInternal()) {
```
Comment (dg) [inline src/DI/ContainerBuilder.php]: `addDependency` is used 3× times (here too https://github.com/nette/di/blob/ea244d0662c164a164bb91e622b00dc361f0668c/src/DI/ContainerBuilder.php#L696), so what about `$this->addDependency((string) $rm->getFileName())` and keep it simple?

File: src/DI/ServiceDefinition.php [hunk 3]
```diff
@@ -48,7 +48,7 @@ class ServiceDefinition extends Nette\Object
 	 */
 	public function setClass($class, array $args = [])
 	{
-		$this->class = ltrim($class, '\\');
+		$this->class = ltrim((string) $class, '\\');
```
Comment (dg) [inline src/DI/ServiceDefinition.php]: `$this->class = $class ? ltrim($class, '\\') : NULL;` seems better here

**Comment by kukulich** [comment]
Force pushed again.

**Comment by dg** [comment]
Great, thanks!

File: src/DI/ContainerBuilder.php [hunk 4]
```diff
@@ -335,7 +335,8 @@ public function prepareClassList()
 		}
 
 		foreach ($this->classes as $class => $foo) {
-			$this->addDependency((new ReflectionClass($class))->getFileName());
+			$rc = new ReflectionClass($class);
```
Comment (dg) [inline src/DI/ContainerBuilder.php]: Ups, there should remain `(new ReflectionClass($class)`… I'll reforce push it.

Comment (kukulich) [inline src/DI/ContainerBuilder.php]: Ok, it was more readable for me so I let it that way.

**Pull request merged**